images/bootstrap/runner.sh

```diff
@@ -41,94 +41,64 @@ setup_ca(){
-        kill "$(</var/run/docker.pid)" || true
-        wait "$(</var/run/docker.pid)" || true
+# runs custom podman data root cleanup binary and debugs remaining resources
+cleanup_pinc() {
+    if [[ "${PODMAN_IN_CONTAINER_ENABLED:-false}" == "true" ]]; then
+        echo "Cleaning up after podman"
+        podman ps -aq | xargs -r podman rm -f || true
+        kill "$(</var/run/podman.pid)" || true
+        wait "$(</var/run/podman.pid)" || true
     fi
 }
 
 early_exit_handler() {
-    cleanup_dind
+    cleanup_pinc
 }
 
 # setup certificates before anything gets started
 setup_ca
 
-# optionally enable ipv6 docker
-export DOCKER_IN_DOCKER_IPV6_ENABLED=${DOCKER_IN_DOCKER_IPV6_ENABLED:-false}
-if [[ "${DOCKER_IN_DOCKER_IPV6_ENABLED}" == "true" ]]; then
-    echo "Enabling IPV6 for Docker."
-    # configure the daemon with ipv6
-    mkdir -p /etc/docker/
-    cat <<EOF >/etc/docker/daemon.json
-{
-  "ipv6": true,
-  "fixed-cidr-v6": "fc00:db8:1::/64"
-}
-EOF
+# optionally enable ipv6
+export PODMAN_IN_CONTAINER_IPV6_ENABLED=${PODMAN_IN_CONTAINER_IPV6_ENABLED:-false}
+if [[ "${PODMAN_IN_CONTAINER_IPV6_ENABLED}" == "true" ]]; then
+    echo "Enabling IPV6."
     # enable ipv6
     sysctl net.ipv6.conf.all.disable_ipv6=0
     sysctl net.ipv6.conf.all.forwarding=1
     # enable ipv6 iptables
     modprobe -v ip6table_nat
 fi
 
-# Check if the job has opted-in to docker-in-docker availability.
-export DOCKER_IN_DOCKER_ENABLED=${DOCKER_IN_DOCKER_ENABLED:-false}
-if [[ "${DOCKER_IN_DOCKER_ENABLED}" == "true" ]]; then
-    echo "Docker in Docker enabled, initializing..."
-
-    export DOCKER_DEBUG=${DOCKER_DEBUG:-false}
-    if [[ "${DOCKER_DEBUG}" == "true" ]]; then
-        mkdir -p /etc/docker/
-        # TODO: do not rely on this file not existing!
-        cat <<EOF >/etc/docker/daemon.json
-{
-  "debug": true,
-  "log-level": "debug"
-}
-EOF
-    fi
-
-    printf '=%.0s' {1..80}; echo
-    # If we have opted in to docker in docker, start the docker daemon,
+export PODMAN_IN_CONTAINER_ENABLED=${PODMAN_IN_CONTAINER_ENABLED:-false}
+if [[ "${PODMAN_IN_CONTAINER_ENABLED}" == "true" ]]; then
+    echo "Podman in Container enabled, initializing in podman compatible mode..."
     (
-        if [ -f "/etc/default/docker" ]; then
-            source /etc/default/docker
-        fi
-        /usr/bin/dockerd \
-            -p /var/run/docker.pid \
-            --data-root=/docker-graph \
-            --init-path /usr/libexec/docker/docker-init \
-            --userland-proxy-path /usr/libexec/docker/docker-proxy \
-            ${DOCKER_OPTS} \
-                >/var/log/dockerd.log 2>&1 &
+        export HTTP_PROXY=${CONTAINER_HTTP_PROXY}
+        export HTTPS_PROXY=${CONTAINER_HTTPS_PROXY}
+	mkdir -p ${XDG_RUNTIME_DIR}/podman
+        podman system service \
+               -t 0 \
+               unix:///${XDG_RUNTIME_DIR}/podman/podman.sock \
+               >/var/log/podman.log 2>&1 &
+        echo "${!}" > /var/run/podman.pid
```

Comment:
I think that instead this we would able to use `systemctl enable --now podman.socket` once https://github.com/kubevirt/kubevirtci/pull/824 is merged (because it knows to detect the default root socket path `/run/podman/socket`)

Reply:
I see thanks

Would it be good idea to include systemd?

What about using the default podman.socket path ? `run/podman/podman.socket`